src/get-started/timestamping-tutorial.md

```diff
@@ -0,0 +1,1017 @@
+
+  ```rust
+  pub struct TimestampProof {
+      /// Proof of the last block.
+      pub block_info: BlockProof,
+      /// Actual state hashes of the timestamping service with their proofs.
+      pub state_proof: MapProof<Hash, Hash>,
+      /// Proof of existence of a specific entry in the timestamping database.
+      pub timestamp_proof: MapProof<Hash, TimestampEntry>,
+  }
+  ```
+
+- the `PublicApi` structure which enables us to implement public API endpoints
+  for our service:
+
+  ```rust
+  pub struct PublicApi;
+  ```
+
+Now we are ready to define the three endpoints required for our service:
+
+```rust
+pub fn handle_post_transaction(
+    state: &ServiceApiState,
+    transaction: TxTimestamp,
+) -> api::Result<Hash> {
+    let hash = transaction.hash();
+    state.sender().send(transaction.into())?;
+    Ok(hash)
+}
+```
+
+The `handle_post_transaction` function sends a timestamping transaction to the
+blockchain performing the following operations:
+
+- take a timestamping transaction
+- calculate its hash
+- send it to the blockchain network, using the methods defined in Exonum
+- return the hash of the transaction to the user.
+
+After the transaction is successfully completed, the system returns the hash
+of the timestamping transaction.
+
+```rust
+pub fn handle_timestamp(
+    state: &ServiceApiState,
+    query: TimestampQuery,
+) -> api::Result<Option<TimestampEntry>> {
+    let snapshot = state.snapshot();
+    let schema = Schema::new(&snapshot);
+    Ok(schema.timestamps().get(&query.hash))
+}
+```
+
+The `handle_timestamp` method takes the hash of a timestamp and
+checks whether a timestamp with this hash exists in the current state of the
+blockchain. To apply this method, the user needs to know the hash of the
+timestamp in question. To perform the check, the method makes a snapshot of
+the current state of the blockchain.
+
+```rust
+pub fn handle_timestamp_proof(
+    state: &ServiceApiState,
+    query: TimestampQuery,
+) -> api::Result<TimestampProof> {
+    let snapshot = state.snapshot();
+    let (state_proof, block_info) = {
+        let core_schema = blockchain::Schema::new(&snapshot);
+        let last_block_height = state.blockchain().last_block().height();
+        let block_proof = core_schema
+            .block_and_precommits(last_block_height).unwrap();
+        let state_proof = core_schema
+            .get_proof_to_service_table(TIMESTAMPING_SERVICE, 0);
+        (state_proof, block_proof)
+    };
+    let schema = Schema::new(&snapshot);
+    let timestamp_proof = schema.timestamps().get_proof(query.hash);
+    Ok(TimestampProof {
+        block_info,
+        state_proof,
+        timestamp_proof,
+    })
+}
+```
+
+`handle_timestamp_proof` takes the hash of a timestamping
+transaction and constructs a proof of its inclusion in the blockchain. To
+construct the proof structure, the method above performs a number of operations
+separated into two blocks:
```

Comment:
use "." instead of ":". Besides I would substitute "block" for "set" or something similar inn order to avoid confusion with the blockchain block